<?xml version="1.0"?>
<launch>
	<!-- topic name -->
	<arg name="ndt_pose_topic_name" default="/ndt_pose" />
	<arg name="imu_topic_name" default="/imu/data" />
	<arg name="ekf_pose_topic_name" default="/ekf_pose" />
	<arg name="odom_topic_name" default="/odom" />
	<arg name="velodyne_topic_name" default="/velodyne_points" />
	<arg name="map_topic_name" default="/global_map" />
	<arg name="ndt_pc_topic_name" default="/ndt_pc" />

	<!-- frame_id -->
	<arg name="map_frame_id" default="map" />
	<arg name="odom_frame_id" default="odom" />
	<arg name="base_link_frame_id" default="base_link" />
	<arg name="camera_frame_id" default="camera_color_optical_frame" />
	<arg name="velodyne_frame_id" default="velodyne" />

	<!-- ekf -->
	<node pkg="rwrc21_recogni_localizer" type="ekf_node" name="ekf_node"  >
		<param name="ndt_pose_topic_name" value="$(arg ndt_pose_topic_name)" />
		<param name="imu_topic_name" value="$(arg imu_topic_name)" />
		<param name="ekf_pose_topic_name" value="$(arg ekf_pose_topic_name)" />
		<param name="odom_topic_name" value="$(arg odom_topic_name)" />
		<param name="map_frame_id" value="$(arg map_frame_id)" />
		<param name="odom_frame_id" value="$(arg odom_frame_id)" />
		<param name="base_link_frame_id" value="$(arg base_link_frame_id)" />
		<param name="is_3DoF" value="true" />
		<param name="is_odom_tf" value="false" />
		<rosparam command="load" file="$(find rwrc21_recogni_localizer)/config/ekf/dkan.yaml" />
	</node>

	<!-- map_matcher -->
	<node pkg="rwrc21_recogni_localizer" type="map_matcher_node" name="map_matcher_node" >
		<!-- <param name="pcd_file_path" value="/home/amsl/pcd_data/d_kan/dkan_1f.pcd" /> -->
		<param name="pcd_file_path" value="/home/amsl/pcd_data/d_kan/d_kan_indoor.pcd" />
		<param name="pc_topic_name" value="$(arg velodyne_topic_name)" />
		<param name="orb_pc_topic_name" value="" />
		<param name="ekf_pose_topic_name" value="$(arg ekf_pose_topic_name)" />
		<param name="ndt_pose_topic_name" value="$(arg ndt_pose_topic_name)" />
		<param name="map_topic_name" value="$(arg map_topic_name)" />
		<param name="ndt_pc_topic_name" value="$(arg ndt_pc_topic_name)" />
		<param name="map_frame_id" value="$(arg map_frame_id)" />
		<rosparam command="load" file="$(find rwrc21_recogni_localizer)/config/map_matcher/dkan.yaml" />
	</node>
	
	<!-- tf -->
	<!-- <node pkg="rwrc21_recogni_localizer" type="tf_broadcaster_node" name="tf_broadcaster_node" >
		<param name="is_odom_tf" value="true" />
		<param name="odom_topic_name" value="$(arg odom_topic_name)" />
		<param name="odom_frame_id" value="$(arg odom_frame_id)" />
		<param name="base_link_frame_id" value="$(arg base_link_frame_id)" />
		<param name="camera_frame_id" value="$(arg camera_frame_id)" />
		<param name="velodyne_frame_id" value="$(arg velodyne_frame_id)" />
		<rosparam command="load" file="$(find rwrc21_recogni_localizer)/config/tf/dkan.yaml" />
	</node> -->

	<!-- rviz -->
	<!-- <node pkg="rviz" type="rviz" name="rviz" args="-d $(find scan_matching)/config/rviz/dkan.rviz" />	 -->

</launch>